forc-plugins/forc-client/src/op/call/list_functions.rs

```diff
@@ -0,0 +1,161 @@
+            .collect::<HashMap<_, _>>();
+
+        for func in &unified_program_abi.functions {
+            let func_args = func
+                .inputs
+                .iter()
+                .map(|input| {
+                    let Ok(param_type) = ParamType::try_from_type_application(input, &type_lookup)
+                    else {
+                        return Err(anyhow!("Failed to convert input type application"));
+                    };
+                    let func_args =
+                        format!("{}: {}", input.name, param_to_function_arg(&param_type));
+                    let func_args_input = {
+                        let token =
+                            param_type_val_to_token(&param_type, &get_default_value(&param_type))
+                                .map_err(|err| {
+                                anyhow!(
+                                    "Failed to generate example call for {}: {}",
+                                    func.name,
+                                    err
+                                )
+                            })?;
+                        token_to_string(&token).map_err(|err| {
+                            anyhow!(
+                                "Failed to convert token to string for {}: {}",
+                                func.name,
+                                err
+                            )
+                        })?
+                    };
+                    Ok((func_args, func_args_input))
+                })
+                .collect::<Result<Vec<_>>>()?;
+
+            let func_args_types = func_args
+                .iter()
+                .map(|(func_args, _)| func_args.to_owned())
+                .collect::<Vec<String>>()
+                .join(", ");
+
+            let func_args_inputs = func_args
+                .iter()
+                .map(|(_, func_args_input)| format!("\"{}\"", func_args_input))
+                .collect::<Vec<String>>()
+                .join(" ");
+
+            let return_type = ParamType::try_from_type_application(&func.output, &type_lookup)
+                .map(|param_type| param_to_function_arg(&param_type))
+                .map_err(|err| {
+                    anyhow!(
+                        "Failed to convert output type application for {}: {}",
+                        func.name,
+                        err
+                    )
+                })?;
+
+            // Get the ABI path or URL as a string
+            let raw_abi_input = match abi {
+                Either::Left(path) => path.to_str().unwrap_or("").to_owned(),
+                Either::Right(url) => url.to_string(),
+            };
+
+            writeln!(
+                writer,
+                "{}({}) -> {}",
+                func.name, func_args_types, return_type
+            )?;
+            writeln!(
+                writer,
+                "  forc call \\\n      --abi {} \\\n      {} \\\n      {} {}\n",
```

Comment:
It might also be good to include the node target arg (if any) provided, so for:

```
forc call   --testnet   --abi https://raw.githubusercontent.com/mira-amm/mira-v1-periphery/refs/heads/main/fixtures/mira-amm/mira_amm_contract-abi.json   0xd5a716d967a9137222219657d7877bd8c79c64e1edb5de9f2901c98ebe74da80 --list-functions
```

The `--testnet` flag would be added to the example call.

Reply:
Will address in another PR; the node-target doesnt impl display; and it has a bunch of other options too.
I'm thinking a better approach may just be a complete pass-through of all the relevant args may be more ideal.